plugins/georeplication/rest.go

```diff
@@ -492,3 +518,314 @@ func georepStatusHandler(w http.ResponseWriter, r *http.Request) {
 	// Send aggregated result back to the client
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, geoSession)
 }
+
+func restartRequiredOnConfigChange(name string) bool {
+	// TODO: Check with Gsyncd about restart required or not
+	// for now restart gsyncd for all config changes
+	return true
+}
+
+func checkConfig(name string, value string) error {
+	args := []string{
+		"config-check",
+		name,
+	}
+	if value != "" {
+		args = append(args, "--value", value)
+	}
+	_, err := exec.Command(gsyncdCommand, args...).Output()
+	return err
+}
+
+func georepConfigGetHandler(w http.ResponseWriter, r *http.Request) {
+	p := mux.Vars(r)
+	masteridRaw := p["mastervolid"]
+	slaveidRaw := p["slavevolid"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	// Validate UUID format of Master and Slave Volume ID
+	masterid, slaveid, err := validateMasterAndSlaveIDFormat(ctx, w, masteridRaw, slaveidRaw)
+	if err != nil {
+		return
+	}
+
+	// Fetch existing session details from Store, error if not exists
+	geoSession, err := getSession(masterid.String(), slaveid.String())
+	if err != nil {
+		if _, ok := err.(*ErrGeorepSessionNotFound); !ok {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+			return
+		}
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "geo-replication session not found", api.ErrCodeDefault)
+		return
+	}
+
+	// Run Local gsyncd and get all configs and its default values
+	args := []string{
+		"config-get",
+		geoSession.MasterVol,
+		fmt.Sprintf("%s@%s::%s", geoSession.SlaveUser, geoSession.SlaveHosts[0], geoSession.SlaveVol),
+		"--show-defaults",
+		"--json",
+	}
+	out, err := exec.Command(gsyncdCommand, args...).Output()
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error":       err.Error(),
+			"mastervolid": masterid,
+			"slavevolid":  slaveid,
+		}).Error("failed to get session configurations")
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Failed to get session configurations", api.ErrCodeDefault)
+		return
+	}
+
+	var opts []georepapi.GeorepOption
+	if err = json.Unmarshal(out, &opts); err != nil {
+		logger.WithFields(log.Fields{
+			"error":       err.Error(),
+			"mastervolid": masterid,
+			"slavevolid":  slaveid,
+		}).Error("failed to parse configurations")
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Failed to parse configurations", api.ErrCodeDefault)
+		return
+	}
+
+	// Apply session configurations
+	for idx, conf := range opts {
+		if val, ok := geoSession.Options[conf.Name]; ok {
+			// Gsyncd returned value is default value
+			if opts[idx].DefaultValue == "" {
+				opts[idx].DefaultValue = opts[idx].Value
```

Comment:
Default value will not be populated from gsyncd, if modified==false. If we found a option in `geoSession.Options` that means it is set by the user. 

gsync will not be aware of the configurations till it is started(Generate config file and start). If Config set and get is requested just after create before start, gsyncd will return default value as empty and modified = false(Since it doesn't have session config file). In this case Value returned by gsyncd is the default value. When georep is started, gsyncd will have access to the session config so it will return default value non empty and modified as true.

Reply:
I will add comment in the code itself.

Reply:
Simplified this code, please review.